### ISSUE TITLE
Errors parsing an iPhone image

### ISSUE THREAD

**Issue by dc3-plaso** [log2timeline/plaso#429]
When I run log2timeline with an iPhone image using the following command:
```
python plaso/tools/log2timeline.py iPhone_out.dump iPhone.img -d --partition all
```

This is a sample of the error output, which looks to be continuous for every file within the image:
```

Processing started.
2015-12-07 13:03:40,456 [DEBUG] (MainProcess) PID:5080 <extraction_frontend> Starting preprocessing.
2015-12-07 13:03:40,542 [WARNING] (MainProcess) PID:5080 <manager> Unable to run preprocessor: MacOSXTimeZone for attribute: time_zone_str with error: Unable to find: /private/etc/localtime
2015-12-07 13:03:40,568 [WARNING] (MainProcess) PID:5080 <manager> Unable to run preprocessor: MacOSXUsers for attribute: users with error: Unable to find user plist files.
2015-12-07 13:03:40,700 [WARNING] (MainProcess) PID:5080 <manager> Unable to run preprocessor: MacOSXBuild for attribute: build with error: No such keys: ProductUserVisibleVersion.
2015-12-07 13:03:40,742 [WARNING] (MainProcess) PID:5080 <manager> Unable to run preprocessor: MacOSXKeyboard for attribute: keyboard_layout with error: Unable to find: /Library/Preferences/com.apple.HIToolbox.plist
2015-12-07 13:03:40,782 [WARNING] (MainProcess) PID:5080 <manager> Unable to run preprocessor: MacOSXHostname for attribute: hostname with error: Unable to find: /Library/Preferences/SystemConfiguration/preferences.plist
2015-12-07 13:03:40,782 [DEBUG] (MainProcess) PID:5080 <extraction_frontend> Preprocessing done.
2015-12-07 13:03:40,782 [INFO] (MainProcess) PID:5080 <extraction_frontend> Parser filter expression changed to: macosx
2015-12-07 13:03:40,782 [DEBUG] (MainProcess) PID:5080 <extraction_frontend> Starting extraction in multi process mode.
2015-12-07 13:03:40,782 [DEBUG] (MainProcess) PID:5080 <multi_process> Starting processes.
WARNING:root:Unable to process path spec: type: OS, location: C:\dev\iPhone.img
type: RAW
type: TSK_PARTITION, location: /p1, part index: 4, start offset: 0x00008000
type: TSK, data stream: DATA, inode: 57154, location: /.file
 with error: Unable to open data stream: DATA.
WARNING:root:Unable to process path spec: type: OS, location: C:\dev\iPhone.img
type: RAW
type: TSK_PARTITION, location: /p1, part index: 4, start offset: 0x00008000
type: TSK, data stream: DATA, inode: 16, location: /.journal
 with error: Unable to open data stream: DATA.
WARNING:root:Unable to process path spec: type: OS, location: C:\dev\iPhone.img
type: RAW
type: TSK_PARTITION, location: /p1, part index: 4, start offset: 0x00008000
type: TSK, data stream: DATA, inode: 17, location: /.journal_info_block
 with error: Unable to open data stream: DATA.
WARNING:root:Unable to process path spec: type: OS, location: C:\dev\iPhone.img
type: RAW
type: TSK_PARTITION, location: /p1, part index: 4, start offset: 0x00008000
type: TSK, data stream: DATA, inode: 60650, location: /^^^^HFS+ Private Data/iNode60650
 with error: Unable to open data stream: DATA.
WARNING:root:Analyzer failed to determine compressed stream type indicators for file: TSK:/^^^^HFS+ Private Data/iNode60650 with error: pysigscan_scanner_scan_file_object: unable to scan file. pysigscan_file_object_read_buffer: unable to read from file object with error: 'File_read_random: (tsk3.c:494) Read error: Invalid file offset (tsk_fs_attr_read - 0)'.
WARNING:root:Analyzer failed to determine archive type indicators for file: TSK:/^^^^HFS+ Private Data/iNode60650 with error: pysigscan_scanner_scan_file_object: unable to scan file. pysigscan_file_object_read_buffer: unable to read from file object with error: 'File_read_random: (tsk3.c:494) Read error: Invalid file offset (tsk_fs_attr_read - 0)'.

...

WARNING:root:Unable to process path spec: type: OS, location: C:\dev\iPhone.img
type: RAW
type: TSK_PARTITION, location: /p2, part index: 5, start offset: 0x55974000
type: TSK, data stream: DATA, inode: 433, location: /wireless/awdd/persistent.db
 with error: Unable to open data stream: DATA.
2015-12-07 13:14:21,790 [DEBUG] (MainProcess) PID:2900 <processing_status> [ProcessingStatus] 120278 pathspecs produced, 119786 processed. 492 remaining. Extraction incomplete.
2015-12-07 13:14:21,790 [DEBUG] (MainProcess) PID:2900 <processing_status> Processing incomplete, extraction still in progress.
2015-12-07 13:14:21,790 [DEBUG] (MainProcess) PID:2900 <processing_status> [ProcessingStatus] 120278 pathspecs produced, 119786 processed. 492 remaining. Extraction incomplete.
Worker_00 (PID: 556) - events extracted: 48128 - file: TSK:/wireless/awdd/staging - running: True <running>
Worker_01 (PID: 3868) - events extracted: 48712 - file: TSK:/wireless/awdd/persistent.db - running: True <running>
Worker_02 (PID: 1084) - events extracted: 47300 - file: TSK:/wireless/spool/MobileOriginated - running: True <running>
Worker_03 (PID: 4008) - events extracted: 46648 - file: TSK:/wireless/spool/MobileOriginated/s.sms.1000000000 - running: True <running>
Worker_04 (PID: 5996) - events extracted: 45976 - file: TSK:/wireless/spool/MobileOriginated/s.sms.1000000000:DATA - running: True <running>
Worker_05 (PID: 4496) - events extracted: 43209 - file: TSK:/wireless/awdd/persistent.db:DATA - running: True <running>
2015-12-07 13:14:30,318 [DEBUG] (MainProcess) PID:2900 <processing_status> Processing incomplete, extraction still in progress.
```

This error appeared around the time data stream support was introduced. Before that, this same image was able to be parsed by log2timeline.

Please let me know if you would like me to pm you more logging information.

**Comment by joachimmetz** [comment]
> Please let me know if you would like me to pm you more logging information.

Thanks for the report. I'll have a look first at HFS data stream handling first, see if I can reproduce it.

**Comment by Onager** [comment]
Adding this as a blocker for the 1.4 release, as this is substantial regression, and seems to be caused by new functionality in 1.4.

**Comment by joachimmetz** [comment]
Reproduced this on a HFS+ image.

**Comment by joachimmetz** [comment]
Testing: http://codereview.appspot.com/279550043

**Comment by joachimmetz** [comment]
This seems to happen when the data fork is named e.g.

```
Attributes: 
Type: DATA (4352-0)   Name: DATA   Resident   size: 4077
Type: CMPF (4355-2)   Name: com.apple.decmpfs   Resident   size: 1598
```

In contrast to:
```
Attributes: 
Type: DATA (4352-0)   Name: N/A   Non-Resident   size: 0  init_size: 0
```

**Comment by joachimmetz** [comment]
CL seems to do the trick for me

**Comment by joachimmetz** [comment]
Should be fixed by dfvfs 20151226, closing issue.
